pkg/pb/gracefulexit.proto

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+syntax = "proto3";
+option go_package = "pb";
+
+import "gogo.proto";
+import "metainfo.proto";
+import "orders.proto";
+
+package gracefulexit;
+
+service SatelliteGracefulExit {
+  // Process is called by storage nodes to initiate the graceful exit, get pieces to transfer, and receive exit status.
+  rpc Process(stream StorageNodeMessage) returns (stream SatelliteMessage);
+}
+
+message TransferSucceeded {
+  metainfo.AddressedOrderLimit addressed_order_limit   = 1;
+  orders.PieceHash             original_piece_hash     = 2;
+  orders.PieceHash             replacement_piece_hash  = 3;
+}
+
+message TransferFailed {
+  bytes piece_id = 1 [(gogoproto.customtype) = "PieceID", (gogoproto.nullable) = false];
+  enum Error {
```

Comment:
would it be okay if this enum were declared outside of the TransferFailed message block? I think that may make it easier to read.

Reply:
actually maybe disregard this, now that I see same patterned is used for Reason below, and it seems fine.